components/brave_wallet/browser/polkadot/polkadot_substrate_rpc.cc

```diff
@@ -367,6 +387,50 @@ void PolkadotSubstrateRpc::OnGetFinalizedHead(GetFinalizedHeadCallback callback,
   return std::move(callback).Run(block_hash, std::nullopt);
 }
 
+void PolkadotSubstrateRpc::GetBlockHeader(
+    std::string_view chain_id,
+    std::optional<base::span<uint8_t, kPolkadotBlockHashSize>> blockhash,
+    GetBlockHeaderCallback callback) {
+  auto url = GetNetworkURL(chain_id);
+
+  base::ListValue params;
+
+  if (blockhash) {
+    params.Append(base::HexEncode(*blockhash));
+  }
+
+  auto payload =
+      base::WriteJson(MakeRpcRequestJson("chain_getHeader", std::move(params)));
+  CHECK(payload);
+
+  api_request_helper_.Request(
+      net::HttpRequestHeaders::kPostMethod, url, *payload, "application/json",
+      base::BindOnce(&PolkadotSubstrateRpc::OnGetBlockHeader,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void PolkadotSubstrateRpc::OnGetBlockHeader(GetBlockHeaderCallback callback,
+                                            APIRequestResult api_result) {
+  auto res =
+      HandleRpcCall<polkadot_substrate_rpc_responses::PolkadotChainHeader>(
+          api_result);
+
+  if (!res.has_value()) {
```

Comment:
it would help if we had a brief comment before each of these failure branches summarising what they are.